pkg/storage/etcd.go

```diff
@@ -137,3 +139,90 @@ func (p *ETCDIPPool) Update(ctx context.Context, reservations []types.IPReservat
 	_, err := p.kv.Put(ctx, fmt.Sprintf("%s/%s", whereaboutsPrefix, p.ipRange), strings.Join(raw, "\n"))
 	return err
 }
+
+// IPManagement manages ip allocation and deallocation from a storage perspective
+func IPManagementEtcd(mode int, ipamConf types.IPAMConfig, containerID string, podRef string) (net.IPNet, error) {
+
+	logging.Debugf("IPManagement -- mode: %v / host: %v / containerID: %v / podRef: %v", mode, ipamConf.EtcdHost, containerID, podRef)
+
+	var newip net.IPNet
+	// Skip invalid modes
+	switch mode {
+	case types.Allocate, types.Deallocate:
+	default:
+		return newip, fmt.Errorf("Got an unknown mode passed to IPManagement: %v", mode)
+	}
+
+	var ipam Store
+	var pool IPPool
+	var err error
+	switch ipamConf.Datastore {
+	case types.DatastoreETCD:
+		ipam, err = NewETCDIPAM(ipamConf)
+	case types.DatastoreKubernetes:
+		ipam, err = NewKubernetesIPAM(containerID, ipamConf)
+	}
```

Comment:
It seems strange to me to find this switch in the `etcd` implementation. Shouldn't the correct config be passed as a parameter to the implementations ?